Tests/FunctionalTests/MiscellaneousTests.swift

```diff
@@ -435,4 +435,21 @@ class MiscellaneousTestCase: XCTestCase {
         }
       #endif
     }
+
+    func testPackageUse() throws {
```

Comment:
If you like my enum approach, we can change this to test the different bin dir search modes. It should also make the test cheaper.

Reply:
I no longer think any part of this is meaningfully testable:

- The search modes can only be tested if there is deterministically something to find. You still haven’t clarified whether that is the case in CI.
- If the search “succeeds” but retrieves something that doesn’t work (say Xcode’s shim, because the validation heuristic is outdated), then the test would be saying the search works when in reality it doesn’t do what it is supposed to.

I suppose we could ignore the result but run the search blindly just to make the tests travel through a few more code paths. But that is about it.